sdk/program/src/sysvar/mod.rs

```diff
@@ -249,6 +249,40 @@ macro_rules! impl_sysvar_get {
     };
 }
 
+/// Handler for retrieving a slice of sysvar data from the `sol_get_sysvar`
+/// syscall.
+fn get_sysvar(
+    dst: &mut [u8],
+    sysvar_id: &Pubkey,
+    offset: u64,
+    length: u64,
+) -> Result<(), ProgramError> {
+    // Check that the provided destination buffer is large enough to hold the
+    // requested data.
+    if dst.len() < length as usize {
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    let sysvar_id = sysvar_id as *const _ as *const u8;
+    let var_addr = dst as *mut _ as *mut u8;
+
+    // Check that the provided destination buffer is aligned to 8.
+    if var_addr.align_offset(8) != 0 {
+        return Err(ProgramError::InvalidArgument);
+    }
```

Comment:
Note that this won't work for every sysvar. For example, the `EpochRewards` sysvar is aligned to 16.

It might be better to leave the alignment check to any helpers that call into this, or as another option, to make this function generic on `T` and do a check like:

```
if var_addr.align_offset(std::mem::align_of::<T>()) != 0
```

But I lean more towards letting people do what they want with this, and having the helper check alignment

Reply:
I'll pull the alignment checks out into the slot hashes caller. Since it's the only caller using the function, I won't impose any alignment within the helper for now.